File: 2024/303. Range Sum Query - Immutable/Solution2.java
class NumArray {

    int[] sums;

    public NumArray(int[] nums) {
        this.sums = new int[nums.length + 1];
        sums[0] = nums[0];
        for (int i = 1; i < nums.length; i++) {
            sums[i] = sums[i - 1] + nums[i];
        }
    }
    
    public int sumRange(int left, int right) {
        if (left == 0)
            return sums[right];
        return sums[right] - sums[left - 1];
    }
}
/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: 2025/271. Encode and Decode Strings/solution2.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        return " & ".join([s.replace("&", "&&") for s in strs])
        
    def decode(self, s: str) -> List[str]:
        return [s.replace("&&", "&") for s in s.split(" & ")]


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))